pkg/cloud/amazon/s3_storage.go

```diff
@@ -141,15 +140,17 @@ func isErrReadConnectionReset(err error) bool {
 	return err != nil && strings.Contains(err.Error(), "read: connection reset")
 }
 
-// ShouldRetry implements the request.Retryer interface.
-func (sr *customRetryer) ShouldRetry(r *request.Request) bool {
-	return sr.DefaultRetryer.ShouldRetry(r) || isErrReadConnectionReset(r.Error)
+// IsErrorRetryable implements the retry.IsErrorRetryable interface.
+func (sr *customRetryer) IsErrorRetryable(e error) aws.Ternary {
+	return aws.BoolTernary(isErrReadConnectionReset(e))
 }
```

Comment:
Potential follow-up: Is necessary anymore? It _looks_ to me like SDK v2 treats connection resets as retriable errors.  https://github.com/aws/aws-sdk-go-v2/blob/7d7814fd914825a22ad375a3f458ae4b81d77ce1/aws/retry/retryable_error.go#L118-L120

Unfortunately we don't have a test around this behaviour.